[CN-127] Add Expose Externally to Hazelcast Enterprise Operator deplo… (#132)
hazelcast/hz-docs: docs/modules/deploy/pages/deploying-with-operator.adoc

NAME                 TYPE           CLUSTER-IP       EXTERNAL-IP     PORT(S)          AGE
hazelcast-sample     LoadBalancer   10.219.246.19    35.230.92.217   5701:30560/TCP   2m11s
hazelcast-sample-0   NodePort       10.219.254.192   <none>          5701:31890/TCP   2m11s
hazelcast-sample-1   NodePort       10.219.247.43    <none>          5701:32310/TCP   2m11s
hazelcast-sample-2   NodePort       10.219.243.16    <none>          5701:32585/TCP   2m11s
----

Then, you use the LoadBalancer External IP of the LoadBalancer service in your Hazelcast client configuration.

Note that you can configure the `expose-externally` feature with the following parameters:

* `type`: the way Hazelcast members are exposed, which implies Hazelcast client type (Unisocket or Smart)
** `Unisocket`: all members are exposed via one Kubernetes service
** `Smart`: each Hazelcast member is exposed with a separate Kubernetes service
* `discoveryServiceType`: Type of Kubernetes service used for discovering Hazelcast cluster
** `LoadBalancer` (default): Hazelcast client configuration `<load-balancer-external-ip>:5701`
** `NodePort`: Hazelcast client configuration `<node-address>:<node-port>`

> * `memberAccess`: the way how each Hazelcast member is accessed

Reviewer: ```suggestion
* `memberAccess`: The way in which clients access Hazelcast members.
```

** `NodePortExternalIP` (default): Hazelcast clients are outside the Kubernetes network, so a client can access members only via the public IP address that's assigned to their Kubernetes nodes.

> ** `NodePortNodeName`: Hazelcast client is inside the same network as the Kubernetes nodes, so members can access Kubernetes nodes by their hostnames

Reviewer: ```suggestion
** `NodePortNodeName`: Hazelcast clients are inside the same network as the Kubernetes nodes, so clients can access a member by the hostname of its Kubernetes node.
```

> ** `NodePortNodeName`: Hazelcast client is inside the same network as the Kubernetes nodes, so members can access Kubernetes nodes by their hostnames

Author: yes
